### PR TITLE
Add partition test cases

### PR DESCRIPTION
**Pull request by ddreimane**
Adding partition testing cases to existing tests and creating new tests. Creating new files for testing like: 
GameRandomTests.java, EntityDistanceComparatorTests.java , NumberAdjusterTests.java etc.

### CODE REVIEW

**Comment by jluech** [comment]
also added a fix to throw an exception in case of division by zero, see [this commit](https://github.com/gurkenlabs/litiengine/pull/382/commits/69f52f5d311e8ba4d956f15c169146a860398db9)

File: tests/de/gurkenlabs/litiengine/GameRandomTests.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class GameRandomTests {
+
+    private GameRandom gameRandom;
+    @BeforeEach
+    public void initSeed(){
+        gameRandom = new GameRandom();
+        gameRandom.setSeed(1337);
```
Comment (nightm4re94) [inline tests/de/gurkenlabs/litiengine/GameRandomTests.java:15]: the only reasonable random seed 😉

File: tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.math.BigDecimal;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import de.gurkenlabs.litiengine.Game;
+import de.gurkenlabs.litiengine.input.Input;
+
+public class NumberAdjusterTests {
```
Comment (nightm4re94) [inline tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java:19]: These changes are also contained in [PR #383](https://github.com/gurkenlabs/litiengine/pull/383/files#diff-3ab447b9083364a5b86a96eb4dd5e406701b43b08b520e519d99a32e203ce64a). Let's hope there won't be any merge conflicts :)

File: tests/de/gurkenlabs/litiengine/resources/FontsTests.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package de.gurkenlabs.litiengine.resources;
+
+import org.junit.jupiter.api.Test;
+
+import java.awt.*;
```
Comment (nightm4re94) [inline tests/de/gurkenlabs/litiengine/resources/FontsTests.java:5]: That's a heavy wildcard import right there :o 
Please use explicit imports instead.

File: tests/de/gurkenlabs/litiengine/util/io/AlphanumComparatorTests.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Comparator;
+
+/**
+ * This is an updated version with enhancements made by Daniel Migowski,
+ * Andre Bogus, and David Koelle. Updated by David Koelle in 2017.
+ *
+ * To use this class:
+ * Use the static "sort" method from the java.util.Collections class:
+ * Collections.sort(your list, new AlphanumComparator());
+ */
```
Comment (nightm4re94) [inline tests/de/gurkenlabs/litiengine/util/io/AlphanumComparatorTests.java:18]: I don't think this comment is intended to be here :)

**Review by nightm4re94 — CHANGES_REQUESTED** [review]
Thank you a lot!
Please address my comments before I can merge this PR.

File: tests/de/gurkenlabs/litiengine/util/io/AlphanumComparatorTests.java (hunk 4 above)
Comment (ddreimane) [reply tests/de/gurkenlabs/litiengine/util/io/AlphanumComparatorTests.java:15]: Very true! Removed

File: tests/de/gurkenlabs/litiengine/resources/FontsTests.java (hunk 3 above)
Comment (ddreimane) [reply tests/de/gurkenlabs/litiengine/resources/FontsTests.java]: Took away the heavy import and changed it to explicit.

File: tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java (hunk 2 above)
Comment (ddreimane) [reply tests/de/gurkenlabs/litiengine/gui/NumberAdjusterTests.java:19]: Mistake from my side. I just kept working with the same file by adding new tests.

**Review by nightm4re94 — APPROVED** [review]

**Comment by Gamebuster19901** [comment]
The package declaration is not correct:

https://github.com/gurkenlabs/litiengine/blob/fdd635e6fcff5081c106d694bd2edb5da6407d1b/tests/de/gurkenlabs/litiengine/util/io/AlphanumComparatorTests.java#L1

**Comment by Gamebuster19901** [comment]
Why is the build succeeding if the declared package is incorrect, and AlphanumComparatorTests is missing an import?

8b95505f9534258687d1e5ee93e8347f178a5d63 83c837d6e40eac96727273b7b907799eadd68d29

**Pull request merged**